src/main/java/net/minecraftforge/client/renderer/transparency/OITLevelRenderer.java

```diff
@@ -0,0 +1,515 @@
+                GameRenderer.getParticleShader() != null && GameRenderer.getParticleShader().OIT_ENABLE != null;
+    }
+
+    /**
+     * Called to start the level rendering process.
+     * Initializes the OIT rendering process.
+     */
+    public void startLevelRendering(ProfilerFiller profilerfiller)
+    {
+        if (isLevelRenderingInProgress())
+            return;
+
+        if (!ForgeConfig.CLIENT.orderIndependentTransparentRendering.get())
+            return;
+
+        profilerfiller.popPush("oit_rendering_setup");
+
+        setLevelRenderingInProgress(true);
+
+        transparentOITRenderTarget.bindWrite(true);
+        glClearBufferfv(GL_COLOR, 0, new float[]{0f, 0f, 0f, 1f});
+        glClearBufferfv(GL_COLOR, 1, new float[]{0f, 0f, 0f, 1f});
+        transparentOITRenderTarget.unbindWrite();
+        Minecraft.getInstance().getMainRenderTarget().bindWrite(true);
+    }
+
+    /**
+     * Called to end the level rendering process.
+     * Finalizes the OIT rendering process.
+     * Should be called immediately after the transparent rendering is done.
+     */
+    public void endLevelRendering(ProfilerFiller profilerfiller)
+    {
+        if (!isLevelRenderingInProgress())
+            return;
+
+        if (!ForgeConfig.CLIENT.orderIndependentTransparentRendering.get())
+            return;
+
+        profilerfiller.popPush("oit_rendering_finalize");
+
+        setLevelRenderingInProgress(false);
+        renderQueue(profilerfiller);
+        queuedRenderCallList.clear();
+    }
+
+    /**
+     * Renders the queued render calls.
+     *
+     * @param profilerfiller The profiler to use.
+     */
+    private void renderQueue(ProfilerFiller profilerfiller) {
+        if (queuedRenderCallList.isEmpty()) {
+            return;
+        }
+
+        profilerfiller.push("oit_rendering_collect");
+
+        transparentOITRenderTarget.copyDepthFrom(Minecraft.getInstance().getMainRenderTarget());
+
+        for (IRenderCall renderCall : queuedRenderCallList) {
+            renderCall.drawOIT();
+        }
+
+        profilerfiller.popPush("oit_rendering_compose");
+
+        transparentOITRenderTarget.unbindWrite();
+
+        transparentOITRenderTarget.blitToScreen(Minecraft.getInstance().getWindow().getWidth(), Minecraft.getInstance().getWindow().getHeight());
+
+        profilerfiller.pop();
+    }
+
+    /**
+     * Callback to set up rendering to the OIT render target.
+     *
+     * @param shaderInstance The shader instance to set up.
+     * @param shouldUpload Whether the uniform values should be uploaded.
+     */
+    private void setupTransparentRendering(ShaderInstance shaderInstance, boolean shouldUpload) {
+        if (shaderInstance.OIT_ENABLE != null) {
+            shaderInstance.OIT_ENABLE.set(1);
+            if (shouldUpload) {
+                shaderInstance.OIT_ENABLE.upload();
+            }
+        }
+        else {
```

Comment:
Bracket should be on a new line/not be present. Realistically it would make more sense to probably just move the body into the if statement and remove the else statement entirely but I am guessing it is done like this so that if we eventually add other types that then the common code is already extracted